### PR TITLE
Message delivery

### PR DESCRIPTION
**Pull request by ghost**
Uses rightmesh library 4.4
mesh manager can now handle delivery report for the messages being sent.
updated the UI to indicate whether messages were delivered or not.
updated database to include delivery column in messages and migrated the database from v4 to v5

### CODE REVIEW

File: app/src/main/java/io/left/meshim/controllers/RightMeshController.java [hunk 1]
```diff
@@ -108,8 +112,10 @@ public void sendTextMessage(User recipient, String message) {
         try {
             byte[] messagePayload = createMessagePayloadFromMessage(messageObject);
             if (messagePayload != null) {
-                meshManager.sendDataReliable(recipient.getMeshId(), MESH_PORT, messagePayload);
-                dao.insertMessages(messageObject);
+                int deliveryDataID = meshManager.sendDataReliable(recipient.getMeshId(), MESH_PORT, messagePayload);
+               long insertedMessageInfo[] = dao.insertMessages(messageObject);
```
Comment (FrazerSeymour) [inline app/src/main/java/io/left/meshim/controllers/RightMeshController.java]: Indentation is too far back one space here and in the line below.

File: app/src/main/java/io/left/meshim/controllers/RightMeshController.java [hunk 2]
```diff
@@ -240,6 +247,8 @@ private void handleDataReceived(RightMeshEvent e) {
 
                 if (sender != null && user != null) {
                     Message message = new Message(sender, user, protoMessage.getMessage(), false);
+                   // message has been delivered
```
Comment (FrazerSeymour) [inline app/src/main/java/io/left/meshim/controllers/RightMeshController.java]: This comment is probably redundant.

Actually, is the `Message` constructor used above only used when creating received messages? If so we can default  delivered to `true` in it and avoid this call.

**Review by FrazerSeymour — APPROVED** [review]

File: app/src/main/java/io/left/meshim/controllers/RightMeshController.java (hunk 1 above)
Comment (ghost) [reply app/src/main/java/io/left/meshim/controllers/RightMeshController.java]: Ah! almost had a perfect PR haha

File: app/src/main/java/io/left/meshim/controllers/RightMeshController.java (hunk 2 above)
Comment (ghost) [reply app/src/main/java/io/left/meshim/controllers/RightMeshController.java]: no it is used in sendTextMessage as well with different value for isMyMessage

File: app/build.gradle [hunk 3]
```diff
@@ -69,6 +69,9 @@ android {
         targetCompatibility 1.8
         sourceCompatibility 1.8
     }
+    configurations.all {
+        resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
+    }
```
Comment (FrazerSeymour) [inline app/build.gradle]: Do we want this in production? I know we've needed it lately due to library stuff on our side but it probably isn't a long-term thing we want.

**Comment by FrazerSeymour** [comment]
Your schema change should have generated another JSON schema file - you'll need to add that.

**Pull request merged**